text/src/autogluon/text/automm/predictor.py

```diff
@@ -566,21 +566,65 @@ def _fit(
 
         if trainer.global_rank == 0:
             top_k_avg_ckpt_path = os.path.join(save_path, "model.ckpt")
+            # Obtain the best K models sorted from the best performance to the worst performance
+            if minmax_mode == 'min':
+                best_k_models_sorted = sorted(list(checkpoint_callback.best_k_models.items()),
+                                              key=lambda ele: ele[1])
+                best_k_models_path = [ele[0] for ele in best_k_models_sorted]
+            elif minmax_mode == 'max':
+                best_k_models_sorted = sorted(list(checkpoint_callback.best_k_models.items()),
+                                              key=lambda ele: ele[1],
+                                              reverse=True)
+                best_k_models_path = [ele[0] for ele in best_k_models_sorted]
```

Comment:
This wasn't present before. Was it bugged previously? Why is this code needed now? Is it because we care about the order of the top K since we aren't simply averaging?

Reply:
Yes, it's because we need to sort these checkpoints based on their performance on the validation set.

Reply:
Also, no bug in the previous version as we are averaging the checkpoints.